x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java

```diff
@@ -613,83 +602,76 @@ public void testCcrRepositoryFailsToFetchSnapshotShardSizes() throws Exception {
+            logger.debug("--> creating follower index [{}]", followerIndex);
+            followerClient().execute(PutFollowAction.INSTANCE, putFollow(leaderIndex, followerIndex, ActiveShardCount.NONE));
 
-            final IndexRoutingTable indexRoutingTable = waitForRestoreInProgress.get(30L, TimeUnit.SECONDS);
+            final List<Long> allShardSnapshotSizes = waitForAllShardSnapshotSizesFailures.get(30L, TimeUnit.SECONDS);
             clusterService.removeListener(listener);
 
-            final SnapshotsInfoService snapshotsInfoService = getFollowerCluster().getCurrentMasterNodeInstance(SnapshotsInfoService.class);
-            assertBusy(() -> {
-                SnapshotShardSizeInfo snapshotShardSizeInfo = snapshotsInfoService.snapshotShardSizes();
-                for (int shardId = 0; shardId < numberOfShards; shardId++) {
-                    final ShardRouting primary = indexRoutingTable.shard(shardId).primaryShard();
-                    assertThat(snapshotShardSizeInfo.getShardSize(primary), equalTo(ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE));
-                    final long randomSize = randomNonNegativeLong();
-                    assertThat(snapshotShardSizeInfo.getShardSize(primary, randomSize), equalTo(randomSize));
-                }
-            }, 60L, TimeUnit.SECONDS);
-        } finally {
-            transportServices.forEach(MockTransportService::clearAllRules);
-        }
+            assertTrue(allShardSnapshotSizes.stream().allMatch(size -> ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE == size));
+            assertThat(simulatedFailures.get(), equalTo(numberOfShards));
 
-        assertThat(indicesStatsRequestsCount.get(), equalTo(numberOfShards));
-        blockCcrRestore.countDown();
+            logger.debug("--> checking that SnapshotsInfoService does not know the real sizes fof snapshot shards");
+            final SnapshotShardSizeInfo snapshotShardSizeInfo = snapshotsInfoService.snapshotShardSizes();
+            for (int shardId = 0; shardId < numberOfShards; shardId++) {
+                final ShardRouting primary = clusterService.state().routingTable().index(followerIndex).shard(shardId).primaryShard();
+                assertThat(snapshotShardSizeInfo.getShardSize(primary), equalTo(ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE));
+                final long randomSize = randomNonNegativeLong();
+                assertThat(snapshotShardSizeInfo.getShardSize(primary, randomSize), equalTo(randomSize));
+            }
 
-        followerClient().admin().cluster().prepareReroute().get();
-        ensureFollowerGreen(followerIndex);
+            if (randomBoolean()) {
+                logger.debug("--> create a random index to generate more cluster state updates");
+                final String randomIndex = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+                assertAcked(followerClient().admin().indices().prepareCreate(randomIndex).setMasterNodeTimeout(TimeValue.MAX_VALUE));
```

Comment:
I wonder if we could use a smaller timeout like 1 minute or so?

Reply:
This timeout is imposed by `CcrIntegTestCase.setupMasterNodeRequestsValidatorOnFollowerCluster()` (see #60070 for more background)

Reply:
:+1: thanks for the clarification.